ZuluContent/Mobiles/PlayerMobile/PlayerMobile.cs

```diff
@@ -945,6 +946,18 @@ public override void SetLocation(Point3D loc, bool isTeleport)
                 RecheckTownProtection();
         }
 
+        public new void Paralyze(TimeSpan duration)
```

Comment:
This should be a ref to a boolean so you can conditionally call `base.Paralyze`/`base.Freeze` based on the side-effect of the hook. When those properties are set on the Mobile they send a delta to the client with the status effect. Setting the duration to 0 essentially sends two deltas instantly telling the client it was frozen/unfrozen which might cause issues if the player is running etc.

e.g.

```csharp
var hookResult = true;
this.FireHook(h => h.OnParalysis(this, ref hookResult));
if(hookResult)
    base.Paralyze(duration);
```

Reply:
```csharp
public new void Paralyze(TimeSpan duration);
public new void Freeze(TimeSpan duration);
```
This a bit of a gotcha and doesn't do what you think it does. The `new` keyword for hiding a base member only hides it when the variable has the type of that class. Meaning that this code will only run in this scenario:

```csharp
// Note the parameter variable type is `PlayerMobile`, the `new` method will be called
public ParalyzeSomeone(PlayerMobile player) { 
    player.Paralyze(TimeSpan.FromSeconds(10.0);
}
```
It won't run here, a live example would be the from the [Paralyze spell itself](https://github.com/ZuluHotelAustralia/zuluhotel/blob/17ba414aeb451f1078ca58cd6c6db4c8d90edb21/ZuluContent/Spells/Fifth/Paralyze.cs#L37)
```csharp
// Now it's Mobile, the base method is not hidden so the base implementation will be called
public ParalyzeSomeone(Mobile player) { 
    player.Paralyze(TimeSpan.FromSeconds(10.0);
}
```

Since the underlying base class method[ `Mobile::Paralyze`](https://github.com/modernuo/ModernUO/blob/b921c38879ca897e90c7c20ca28c674cf09b922d/Projects/Server/Mobiles/Mobile.cs#L3829) is not `virtual` we can't override it. The fix is would be to get an upstream PR to ModernUO merged that makes `Freeze` and `Paralyze` virtual methods.

Reply:
Ah thanks for explaining, I was thinking exactly that might be the case actually but didn't get a chance to test whether this would be called or not in the game.

Will make a PR to ModernUO with that change then and rather use the boolean way, makes a lot more sense actually.

Reply:
Updated this now